public/data.js

```diff
@@ -0,0 +1,54 @@
+class DataController {
+    static updateData(snapshot, data) {
+        // Update data every time it changes in firestore
+        snapshot.docChanges().forEach(change => {
+
+            const doc = { ...change.doc.data(), id: change.doc.id };
+
+            switch (change.type) {
+                case "added":
+                    data.push(doc);
+                    break;
+                case "modified":
+                    const index = data.findIndex(item => item.id == doc.id);
+                    data[index] = doc;
+                    break;
+                case "removed":
+                    data = data.filter(item => item.id !== doc.id);
+                    break;
+                default:
+                    break;
+            }
+        });
+    }
+
+    static watchActiveProjects(onChange) {
+        let activeProjects = []
+        mediadb.collection('active_projects').onSnapshot(res => {
+            // Update data every time it changes in firestore
+            DataController.updateData(res, activeProjects);
+            onChange(activeProjects);
+        });  
+    }
+
+    static watchCodingProgress(onChange) {
+        mediadb.doc('metrics/coda').onSnapshot(res => {
+            onChange(res.data());
+        });
+    }
+
+    static watchProjectData(projectName, onChange) {  
```

Comment:
project coding data or project traffic data?

Suggest renaming this to 'watchProjectTrafficData' or similar

Reply:
Done